src/couch_index/src/couch_index_server.erl

```diff
@@ -322,15 +336,47 @@ handle_db_event(<<"shards/", _/binary>> = DbName, {ddoc_updated, DDocId}, St) ->
 handle_db_event(DbName, {ddoc_updated, DDocId}, St) ->
     lists:foreach(
         fun({_DbName, {_DDocId, Sig}}) ->
-            case ets:lookup(?BY_SIG, {DbName, Sig}) of
+            case ets:lookup(St#st.by_sig, {DbName, Sig}) of
                 [{_, IndexPid}] ->
                     (catch gen_server:cast(IndexPid, ddoc_updated));
                 [] ->
                     ok
             end
         end,
-        ets:match_object(?BY_DB, {DbName, {DDocId, '$1'}})
+        ets:match_object(St#st.by_db, {DbName, {DDocId, '$1'}})
     ),
     {ok, St};
 handle_db_event(_DbName, _Event, St) ->
     {ok, St}.
+
+num_servers() ->
+    erlang:system_info(schedulers).
+
+couch_index_server(Arg) ->
+    name("index_server", Arg).
+
+by_sig(Arg) ->
+    name("couchdb_indexes_by_sig", Arg).
+
+by_pid(Arg) ->
+    name("couchdb_indexes_by_pid", Arg).
+
+by_db(Arg) ->
+    name("couchdb_indexes_by_db", Arg).
+
+name(BaseName, Sig) when is_list(Sig) ->
+    name(BaseName, ?l2b(Sig));
+name(BaseName, Sig) when is_binary(Sig) ->
+    N = 1 + erlang:phash2(Sig, num_servers()),
+    name(BaseName, N);
+name(BaseName, N) when is_integer(N), N > 0 ->
+    list_to_atom(BaseName ++ "_" ++ integer_to_list(N)).
+
+aggregate_queue_len() ->
+    N = num_servers(),
+    Names = [couch_index_server(I) || I <- lists:seq(1, N)],
+    MQs = [
+        process_info(whereis(Name), message_queue_len)
+     || Name <- Names
```

Comment:
If there is a chance stats would ever be queried before indexing servers have been started, this might crash. `whereis/1` would return `undefined` and process_info would throw a badarg. 

If you think it's worth can gather Pids instead of name, and then, in the MQs expression filter out `undefined`

Reply:
the couch_index_server_X processes are created in secondary_sup, I don't think we need worry about calls to _system before that happens.

Reply:
Hmm, it could matter if we crash the stats reporter when we also have a crashing index server. Normally we'd want not want to be defensive but when reporting stats it might make sense...